src/Product/Single.php

```diff
@@ -190,7 +190,7 @@ public function save_additional_product_shipping_options( $product_id, $i = '' )
 		$fields = self::product_field_maps( $this->service );
 
 		foreach ( $fields as $field ) {
-			if ( empty( $i ) && ! is_array( $_POST[ $field['id'] ] ) && 0 === $product->get_parent_id() ) {
+			if ( empty( $i ) && isset( $_POST[ $field['id'] ] ) && ! is_array( $_POST[ $field['id'] ] ) && 0 === $product->get_parent_id() ) {
```

Comment:
Lets simplify the code and return first if its not exist and also, if the key is not exist we should delete the meta data, for example :
the letterbox checkbox, if its checked the POST will contains a value 'YES' for the `_postnl_letterbox_parcel` key, but if its not checked it will not contains the key at all unless its a variations in this case it will be empty value, so this logic will just ignore it and the user will never be able to uncheck the field


```suggestion
			if ( ! isset( $_POST[ $field['id'] ] ) ) {
				$product->delete_meta_data( $field['id'] );
				continue;
			}

			if ( empty( $i ) && ! is_array( $_POST[ $field['id'] ] ) && 0 === $product->get_parent_id() ) {
```